### PR TITLE
[#319] Refactor add credit module configuration.

### LINKED ISSUE #319: "Unable to install Apigee Kickstart Monetization due to unmet dependencies: field.field.commerce_product.add_credit.stores (field.storage.commerce_product.stores), field.field.commerce_product.add_credit.variations (field.storage.commerce_product.variations)" error
I have the D8 distro installed with Apigee Edge configured. When I try to enable the "Apigee Kickstart Monetization" experimental feature I get the following warning:

"Unable to install Apigee Kickstart Monetization due to unmet dependencies: field.field.commerce_product.add_credit.stores (field.storage.commerce_product.stores), field.field.commerce_product.add_credit.variations (field.storage.commerce_product.variations)" error

My guess is that this configuration is supposed to be provided by other modules in the project such as "Apigee Monetization Add Credit" which is enabled.

### PR DESCRIPTION
**Pull request by arlina-espinoza**
This PR needs https://github.com/apigee/apigee-m10n-drupal/pull/223 (Compatibility with commerce 8.x-2.16), and together fix #319 .
The `Apigee Kickstart Monetization` submodule of Kickstart (`apigee_kickstart_m10n_add_credit`) is a configuration only module, and it is meant to be enabled during the initial Drupal installation, in the monetization step if the _"Enable Add Credit module"_ is checked. If so, the Kickstart installer provided supplemental initialization in code. If the module is enabled _after_ the installer completed, the other logic isn't run. 
To solve this issue, efforts were made in the `apigee_m10n` module to allow a simpler configuration post-installation, and `apigee_m10n 8.x-1.3` includes a way to configure the "Add credit" module in a simple way from a _requirements_ page at: `/admin/reports/requirements`
- This PR leverages that new feature in `apigee_m10n` during the install phase if _"Enable Add Credit module"_ is checked.
- This PR also deprecates the `apigee_kickstart_m10n_add_credit` submodule, as it's no longer needed, and removes the confusion of having a module that should not be installed _after_ the initial installation.
- We need a follow up ticket to call attention to the `apigee_m10n_add_credit` configuration page at: `/admin/reports/requirements`

@drupalninja: To test:
- You need this PR and https://github.com/apigee/apigee-m10n-drupal/pull/223
- No need to enable the `Apigee Kickstart Monetization` module. Just enable `apigee_m10n_add_credit` (Apigee Monetization Add Credit), and complete the requirements listed under `/admin/reports/requirements`

### CODE REVIEW

File: modules/custom/apigee_kickstart_m10n_add_credit/apigee_kickstart_m10n_add_credit.info.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name: Apigee Kickstart Monetization
-description: "Handles monetization for Apigee Kickstart. Adds and configures a default product type for adding credit."
+name: Apigee Kickstart Monetization (Deprecated)
```
Comment (cnovak) [inline modules/custom/apigee_kickstart_m10n_add_credit/apigee_kickstart_m10n_add_credit.info.yml]: In D7, you could hide modules, is that something we should do here?

Comment (cnovak) [inline modules/custom/apigee_kickstart_m10n_add_credit/apigee_kickstart_m10n_add_credit.info.yml:1]: In D7, you could hide modules, is that something we should do here?

File: apigee_devportal_kickstart.install [hunk 2]
```diff
@@ -150,6 +150,10 @@ function apigee_devportal_setup_monetization(array &$install_state) {
           [ApigeeDevportalKickstartTasksManager::class, 'createStore'],
           [$config['store']],
         ],
+        [
+          [ApigeeDevportalKickstartTasksManager::class, 'createProductType'],
```
Comment (cnovak) [inline apigee_devportal_kickstart.install:154]: `ApigeeDevportalKickstartTasksManagerInterface` has a method called `createPaymentGateway`, why is it not being called?

File: src/Installer/ApigeeDevportalKickstartTasksManager.php [hunk 3]
```diff
@@ -97,6 +98,27 @@ public static function createPaymentGateway(array $values, array &$context) {
       $context['results']['$gateway'] = $gateway;
       $context['message'] = t('Created a default payment gateway.');
     }
+    catch (\Exception $exception) {
```
Comment (cnovak) [inline src/Installer/ApigeeDevportalKickstartTasksManager.php:106]: See message above, `createPaymentGateway` is not being called, but should be

**Review by cnovak — CHANGES_REQUESTED** [review]
Added comments. Do we have instructions/issue to track doc changes that need to be done? 
I believe we need:

1. Kickstart docs: Update instructions for setting up monetization during installer
2. Kickstart docs: Update/create how to install monetization after installing if not done during installer
3. M10n docs: Update/create Steps needed to configure m10n 

Can we add a ticket w/this info?

File: modules/custom/apigee_kickstart_m10n_add_credit/apigee_kickstart_m10n_add_credit.info.yml (hunk 1 above)
Comment (arlina-espinoza) [reply modules/custom/apigee_kickstart_m10n_add_credit/apigee_kickstart_m10n_add_credit.info.yml:1]: Yep, I've added `hidden: true` to it's yml file.

File: apigee_devportal_kickstart.install (hunk 2 above)
Comment (arlina-espinoza) [reply apigee_devportal_kickstart.install:154]: I've added back this step, with a the option to disable it in the monetization setup form:
image

File: src/Installer/ApigeeDevportalKickstartTasksManager.php (hunk 3 above)
Comment (arlina-espinoza) [reply src/Installer/ApigeeDevportalKickstartTasksManager.php:106]: Fixed :)

**Comment by arlina-espinoza** [comment]
@cnovak I've addressed the changes, could you review again please?

File: apigee_devportal_kickstart.install (hunk 2 above)
Comment (cnovak) [reply apigee_devportal_kickstart.install:154]: nice

**Comment by arlina-espinoza** [comment]
Added documentation issues: https://github.com/apigee/apigee-m10n-drupal/issues/226 and https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/330.

**Review by cnovak — APPROVED** [review]
LGTM

**Pull request merged**